### PR TITLE
Customize all the poke filters

### PR DESCRIPTION
**Pull request by sketchysocks**
- Moved rarity tiers to config to be customizable

- Renamed rarity.json to generation.json, since that's all it is now

- Cleaned up Quick Start code to be all quick start related, not recommended & quick start 😂

- Changed default IV filters to be 90-100 and PVP 1-3, because users need "help"

- Removed Alola/Galars from generation.json since they aren't necessary

- Fixed Galarian Darmanitan from activating when using the Galarian filter buttons

### CODE REVIEW

File: src/data/default.js [hunk 1]
```diff
@@ -31,6 +31,12 @@ data.default_show_submissioncells = config.map.filters.submissionCells;
...
 data.default_show_scanareas = config.map.filters.scanAreas;
 data.default_show_devices = config.map.filters.devices;
-data.recommended_pokemon_json = JSON.stringify(config.recommendedPokemon);
+data.quickStart_pokemon_json = JSON.stringify(config.pokemon.quickStart);
+data.common_pokemon_json = JSON.stringify(config.pokemon.common);
+data.uncommon_pokemon_json = JSON.stringify(config.pokemon.uncommon);
+data.rare_pokemon_json = JSON.stringify(config.pokemon.rare);
+data.ultraRare_pokemon_json = JSON.stringify(config.pokemon.ultraRare);
+data.regional_pokemon_json = JSON.stringify(config.pokemon.regional);
+data.event_pokemon_json = JSON.stringify(config.pokemon.event);
```
Comment (Mygod) [inline src/data/default.js:40]: Why not just use a single field for the entire thing?

File: src/views/index-js.mustache [hunk 2]
```diff
@@ -137,7 +137,14 @@ let defaultShowNests = '{{default_show_nests}}' === 'true';
...
 
-const recommendedPokemon = JSON.parse('{{{recommended_pokemon_json}}}');
+    
+const quickStartPokemon = JSON.parse('{{{quickStart_pokemon_json}}}');
+const commonPokemon = JSON.parse('{{{common_pokemon_json}}}');
+const uncommonPokemon = JSON.parse('{{{uncommon_pokemon_json}}}');
+const rarePokemon = JSON.parse('{{{rare_pokemon_json}}}');
+const ultraRarePokemon = JSON.parse('{{{ultraRare_pokemon_json}}}');
+const regionalPokemon = JSON.parse('{{{regional_pokemon_json}}}');
+const eventPokemon = JSON.parse('{{{event_pokemon_json}}}');
```
Comment (Mygod) [inline src/views/index-js.mustache:147]: Same thing here.

**Comment by versx** [comment]
Anyone test latest changes?

**Comment by versx** [comment]
Works well, just need to add `rarity.json` to the `.gitignore` file and it's ready to merge.

**Comment by sketchysocks** [comment]
> Works well, just need to add `rarity.json` to the `.gitignore` file and it's ready to merge.

Done 👌

**Pull request merged**